### PR TITLE
DSA-23-Longest-Palindromic-String

### PR DESCRIPTION
**Pull request by ganeshyevle**
Given a String "S", find the longest palindromic substring in"S

 Input "S" ="BABAD"

 Output="BAB"

### CODE REVIEW

File: DSA_Training/longestPalindromicSubstring.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+function longestPalindromicSubstring(s) {
```
Comment (Umarfarook1) [inline DSA_Training/longestPalindromicSubstring.js:2]: please add proper comment

File: DSA_Training/longestPalindromicSubstring.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    if (!s) return "";
+
```
Comment (Umarfarook1) [inline DSA_Training/longestPalindromicSubstring.js]: please remove extra space

File: DSA_Training/longestPalindromicSubstring.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    let longest = "";
+
+    function expandAroundCenter(left, right) {
```
Comment (Umarfarook1) [inline DSA_Training/longestPalindromicSubstring.js:7]: please add comment for the function

File: DSA_Training/longestPalindromicSubstring.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        let evenPalindrome = expandAroundCenter(i, i + 1);
+        if (evenPalindrome.length > longest.length) {
+            longest = evenPalindrome;
+        }
+    }
+
+    return longest;
+}
+
+// Example usage:
```
Comment (Umarfarook1) [inline DSA_Training/longestPalindromicSubstring.js]: please remove example usage code

File: DSA_Training/longestPalindromicSubstring.js (hunk 1 above)
Comment (ganeshyevle) [reply DSA_Training/longestPalindromicSubstring.js:2]: done

File: DSA_Training/longestPalindromicSubstring.js (hunk 3 above)
Comment (ganeshyevle) [reply DSA_Training/longestPalindromicSubstring.js:7]: done

File: DSA_Training/longestPalindromicSubstring.js (hunk 4 above)
Comment (ganeshyevle) [reply DSA_Training/longestPalindromicSubstring.js]: done

File: DSA_Training/longestPalindromicSubstring.js (hunk 2 above)
Comment (ganeshyevle) [reply DSA_Training/longestPalindromicSubstring.js]: done

**Review by Umarfarook1 — APPROVED** [review]

**Pull request merged**